pkg/cluster/cluster.go

```diff
@@ -215,6 +215,12 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 
 	var rerr error
 	for {
+		if isFatalError(rerr) {
```

Comment:
it is less readable. probably we should still move this to the end of the for loop.